### PR TITLE
ENT-3793, ENT-3795, ENT-3797: Updated completion, course structure and grade reports for file download from S3

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-analytics-data-api doesn't install it
    - `test-master.in` if edx-analytics-data-api pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the edx-analytics-data-api requirements installed or if you checked out edx-enterprise-data into the src directory used by edx-analytics-data-api, you can run this command through an edx-analytics-data-api shell.
        - It would be `./manage.py makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise-data/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise-data/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise-data/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise-data), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise-data/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-analytics-data-api](https://github.com/edx/edx-analytics-data-api) to upgrade dependencies (including edx-enterprise-data)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-analytics-data-api will look for the latest version in PyPi.
    - Note: the edx-enterprise-data constraint in edx-analytics-data-api **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise_reporting/clients/s3.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+class S3Client:
+    """
+    Client for connecting to AWS S3.
+    """
+
+    def __init__(self):
+        pass
+
+    def get_file(self, S3_object_name, file_handler):
```
Comment (saleem-latif) [inline enterprise_reporting/clients/s3.py]: nit: I think we should use snake case here i.e. rename to `s3_object_name `

File: enterprise_reporting/clients/s3.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        """
+        Download a file from S3
```
Comment (saleem-latif) [inline enterprise_reporting/clients/s3.py]: nit, add argument details in the doc string.

```
Arguments:
    s3_object_name (<type>): <description>
    file_handler (<type>): <description>
```

File: enterprise_reporting/reporter.py [hunk 3]
```diff
@@ -394,18 +394,10 @@ def _generate_enterprise_report_course_structure_csv(self):
...
-            query = self.SNOWFLAKE_COMPLETION_QUERY.format(
-                fields=','.join(self.SNOWFLAKE_COMPLETION_QUERY_FIELDS),
-                enterprise_id=self.enterprise_customer_uuid.replace('-', '')
-            )
-            LOGGER.debug('Executing this Snowflake query: {}'.format(query))
-            data_report_file_writer.writerows(snowflake_client.stream_results(query))
-        snowflake_client.close_connection()
+        s3_client = S3Client()
+        with open(self.data_report_file_name, 'wb') as data_report_file:
+            LOGGER.debug('Fetching enterprise completion report from S3)
```
Comment (saleem-latif) [inline enterprise_reporting/reporter.py]: would it be helpful to add more information in the debug message? like bucket name and target file name?

**Review by saleem-latif — APPROVED** [review]
A couple of comments otherwise looks good 👍

File: enterprise_reporting/clients/s3.py (hunk 1 above)
Comment (irfanuddinahmad) [reply enterprise_reporting/clients/s3.py]: Done

File: enterprise_reporting/clients/s3.py (hunk 2 above)
Comment (irfanuddinahmad) [reply enterprise_reporting/clients/s3.py]: Done

File: enterprise_reporting/reporter.py (hunk 3 above)
Comment (irfanuddinahmad) [reply enterprise_reporting/reporter.py]: 'completion' is the name of the file. We don't have bucket info here.

**Review by muhammad-ammar — APPROVED** [review]

**Pull request merged**